resdk/resolwe.py

```diff
@@ -508,20 +512,26 @@ def _download_files(self, files, download_dir=None):  # pylint: disable=redefine
 class ResAuth(requests.auth.AuthBase):
     """HTTP Resolwe Authentication for Request object.
 
-    :param email: user's email
-    :type email: str
-    :param password: user's password
-    :type password: str
-    :param url: Resolwe server instance
-    :type url: str
+    :param str username: user's username
+    :param str password: user's password
+    :param str url: Resolwe server instance
 
     """
 
-    def __init__(self, email=DEFAULT_EMAIL, password=DEFAULT_PASSWD, url=DEFAULT_URL):
+    sessionid = None
+    csrftoken = None
+
+    def __init__(self, username=None, password=None, url=DEFAULT_URL):
         """Authenticate user on Resolwe server."""
         self.logger = logging.getLogger(__name__)
 
-        payload = {'username': email, 'password': password}
+        self.username = username
+        self.url = url
+
+        if not username and not password:
```

Comment:
It works, if you make request without cookies, public user is automatically assigned on the server.